### PR TITLE
Fix #8073: Correctly escape '#' character in key & label paths and queries

### LINKED ISSUE #8073: Add tests to validate: correctly escape '#' character in key & label paths and queries
This call will fail with a 401 error:

`await service.SetAsync(new ConfigurationSetting("name-!@#$^&()-=", "value-!@#$^&()-=", "label-!@#$^&()-="));`

This is due to improper escaping of key in the URI path, and label in the query parameter.

All client methods that take key and label should handle the '#' character correctly.

### PR DESCRIPTION
**Pull request by AlexanderSher**
The bug is actually fixed, this PR adds the test to for this issue.

### CODE REVIEW

File: sdk/appconfiguration/Azure.Data.AppConfiguration/tests/ConfigurationLiveTests.cs [hunk 1]
```diff
@@ -874,6 +874,37 @@ public async Task GetBatchSettingWithAllFields()
             }
         }
 
+        [TestCase(true)]
+        [TestCase(false)]
```
Comment (annelo-msft) [inline sdk/appconfiguration/Azure.Data.AppConfiguration/tests/ConfigurationLiveTests.cs]: The convention for our tests is to use the [Test] attribute on test methods.  Is there a reason to do it differently here?

File: sdk/appconfiguration/Azure.Data.AppConfiguration/tests/ConfigurationLiveTests.cs [hunk 2]
```diff
@@ -874,6 +874,37 @@ public async Task GetBatchSettingWithAllFields()
...
+                var selector = new SettingSelector(key, label);
+
+                Assert.AreEqual(key, selector.Keys.First());
+                Assert.AreEqual(label, selector.Labels.First());
+
+                var resultsReturned = (await service.GetSettingsAsync(selector, CancellationToken.None).ToEnumerableAsync());
+
+                //At least there should be one key available
+                Assert.GreaterOrEqual(resultsReturned.Count, 1);
+                Assert.AreEqual(value, resultsReturned[0].Value);
```
Comment (annelo-msft) [inline sdk/appconfiguration/Azure.Data.AppConfiguration/tests/ConfigurationLiveTests.cs]: Let's add a test to verify the equality of key/Key here as well.

File: sdk/appconfiguration/Azure.Data.AppConfiguration/tests/ConfigurationLiveTests.cs [hunk 3]
```diff
@@ -874,16 +873,52 @@ public async Task GetBatchSettingWithAllFields()
...
+            {
+                await service.DeleteAsync(testSetting.Key, testSetting.Label);
+            }
+        }
+
+        [TestCase(true, true)]
+        [TestCase(false, true)]
+        [TestCase(true, false)]
+        [TestCase(false, false)]
+        public async Task GetBatchSettingComma(bool useWildcardForKey, bool useWildcardForValue)
```
Comment (annelo-msft) [inline sdk/appconfiguration/Azure.Data.AppConfiguration/tests/ConfigurationLiveTests.cs]: Could you add one for commas with escaped characters as well?

**Comment by AlexanderSher** [comment]
Closing in favor of #8264

**Pull request closed without merging**